include/yaml-cpp/node/convert.h

```diff
@@ -114,6 +115,44 @@ typename std::enable_if<!std::is_floating_point<T>::value, void>::type
 inner_encode(const T& rhs, std::stringstream& stream){
   stream << rhs;
 }
+
+template <typename T>
+typename std::enable_if<std::is_same<T, unsigned char>::value, bool>::type
+ConvertStreamTo(std::stringstream& stream, T& rhs) {
+  uint16_t num;
```

Comment:
Why only uint16_t? That means that 2^16+1 will be parsed at the whims of std::stringstream. Speaking of which, what would happen in that case?

Reply:
I have updated the `ConvertStreamTo()` . If the stream is over the range of `int8_t/uint8_t`. It will throw an exception. And, the uint16_t is a transit data type for uint8_t/int8_t.  There are many options for this and I have change it to `int`.